src/layout/engagement_details_view.tsx

```diff
@@ -74,7 +75,11 @@ export function EngagementDetailsViewTemplate({
             <FlexItem>
               <TextContent>
                 <Text component="h3" style={{ marginTop: '0.5rem' }}>
-                  {engagement?.engagement_type}
+                  {
+                    engagementFormConfig?.basic_information?.engagement_types?.options?.find?.(
```

Comment:
use this helper function: `src/common/label_value_tools.ts`

Reply:
Applied your feedback. Thank you.